clients/src/main/java/org/apache/kafka/common/metrics/KafkaMetric.java

```diff
@@ -20,6 +20,37 @@
 import org.apache.kafka.common.MetricName;
 import org.apache.kafka.common.utils.Time;
 
+/**
+ * An implementation of {@link Metric} interface.
+ * <p>
+ * A KafkaMetric is a named metric for monitoring purpose. The metric value can be a {@link Measurable} or a {@link Gauge}.
+ * <pre>
+ * <b>metricName</b> The name of the metric
+ * <b>lock</b> A lock used for reading the metric value in case of race condition
+ * <b>time</b> The POSIX time in milliseconds the metric is being taken
+ * <b>metricValueProvider</b> The metric collecting implementation that implements {@link MetricValueProvider}
+ * <b>config</b> The metric configuration which is a {@link MetricConfig}
+ * </pre>
+ * <p>
+ * Usage looks something like this:
+ *
+ * <pre>{@code
+ * // set up metrics:
+ *
+ * Map<String, String> tags = new HashMap<>();
+ * tags.put("key1", "value1");
+ *
+ * MetricConfig config = new MetricConfig().tags(metricTags);
+ * Time time = new SystemTime();
+ * metricName = new MetricName(&quot;message-size-max&quot;, &quot;producer-metrics&quot;);
+ *
+ * KafkaMetric m = new KafkaMetric(new Object(),
```

Comment:
It seems to me the point of usage is how to use its public methods rather than how to build this object. The reason of making `KafkaMetrics` be a public Api is that `MetricsReporter` (https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/metrics/MetricsReporter.java#L39), which is a public interface, has many methods taking `KafkaMetric` as argument.

Reply:
thanks for the comment! yes that makes sense to me, I think I missed the point.
Just updated and added doc for public methods, thanks!